tests/marine/benchmark/utils.py

```diff
@@ -0,0 +1,54 @@
+import struct
+from dataclasses import dataclass, field
+from random import randint
+from typing import List, Dict, Callable, Set, Tuple
+
+
+@dataclass
+class BenchmarkPacket:
+    packet: bytes
+    good_bpf: str
+    good_display_filter: str
+    # TODO: add bad_bpf and bad_display_filter to drop some packets
+    # bad_bpf: str
+    # bad_display_filter: str
+    fields_to_extract: List[str]
+    expected_parse_result: Dict[str, str]
+
+
+@dataclass
+class IpPair:
+    src_mac: str
+    dst_mac: str
+    src_ip: str
+    dst_ip: str
+    _ports: Set[int] = field(default_factory=set)
+
+    def _generate_port(self) -> int:
```

Comment:
This ugly stateful logic really shouldn't reside here

Reply:
How would you suggest doing it? I want a unique port set per `IpPair`